yt_dlp/extractor/digitalconcerthall.py

```diff
@@ -0,0 +1,201 @@
+    def debug_out(self, args):
+        if not self._downloader.params.get('verbose', False):
+            return
+
+        self.to_screen('[debug] %s' % args)
+
+    def _login(self):
+        username, password = self._get_login_info()
+        if username is None:
+            if self._LOGIN_REQUIRED:
+                raise ExtractorError('No login info available, needed for using %s.' % self.IE_NAME, expected=True)
+            return
+        # first get JWT token
+        data = {
+            'affiliate': 'none',
+            'grant_type': 'device',
+            'device_vendor': 'unknown',
+            'device_model': 'unknown',
+            'app_id': 'dch.webapp',
+            'app_distributor': 'berlinphil',
+            'app_version': '1.0.0',
+            'client_secret': self._CLIENT_SECRET,
+        }
+        try:
+            token_response = self._download_json(
+                self._OAUTH_URL,
+                None, 'Obtaining token', data=urlencode_postdata(data), headers={
+                    'Content-Type': 'application/x-www-form-urlencoded',
+                })
+        except ExtractorError as e:
+            msg = 'Unable to obtain token'
+            if isinstance(e.cause, compat_HTTPError) and e.cause.code == 401:
+                resp = self._parse_json(e.cause.read().decode(), None, fatal=False)
+                if resp:
+                    error = resp.get('extra_info') or resp.get('error_description') or resp.get('error')
+                    if error:
+                        msg += ': ' + error
+            raise ExtractorError('Unable to obtain token: ' + msg)
+        self.debug_out("token_response: " + json.dumps(token_response))
+        self._ACCCESS_TOKEN = token_response.get('access_token')
+        # now login
+        data = {
+            'grant_type': 'password',
+            'username': username,
+            'password': password,
+        }
+        try:
+            self._download_json(
+                self._OAUTH_URL,
+                None, 'Logging in', data=urlencode_postdata(data), headers={
+                    'Content-Type': 'application/x-www-form-urlencoded',
+                    'Referer': self._LOGIN_URL,
+                    'Authorization': 'Bearer ' + self._ACCCESS_TOKEN
+                })
+        except ExtractorError as e:
+            msg = 'Unable to login'
+            if isinstance(e.cause, compat_HTTPError) and e.cause.code == 401:
+                resp = self._parse_json(e.cause.read().decode(), None, fatal=False)
+                if resp:
+                    error = resp.get('extra_info') or resp.get('error_description') or resp.get('error')
+                    if error:
+                        msg += ': ' + error
+            raise ExtractorError('Unable to log in: ' + msg)
+
+    def _real_initialize(self):
+        self._login()
+
+    def _real_extract(self, url):
+        MAX_TITLE_LENGTH = 128
+        language, video_id = re.match(self._VALID_URL, url).groups()
```

Comment:
use `_match_valid_url` instead of `re.match`

Reply:
see 5d4feb232e3c9a34e154d829600da8f962d60be7